### PR TITLE
Fail the try_step if the inner step was aborted [#3989].

### PR DESCRIPTION
**Pull request by josecv**
Signed-off-by: jose <jose@autonomic.ai>

### CODE REVIEW

**Comment by josecv** [comment]
I'm not a golang understander so please let me know if I've made any mistakes, I'll be glad to fix em up

**Comment by xtremerui** [comment]
@vito the fix is good to me. However since there is a nice example in the issue that could reproduces the problem, is it worth to add a topgun test based on that? Could be a separate PR.

File: atc/exec/try_step.go [hunk 1]
```diff
@@ -7,20 +7,23 @@ import (
 // TryStep wraps another step, ignores its errors, and always succeeds.
 type TryStep struct {
 	step Step
+	aborted bool
 }
 
 // Try constructs a TryStep.
 func Try(step Step) Step {
-	return TryStep{
+	return &TryStep{
```
Comment (pivotal-jwinters) [inline atc/exec/try_step.go]: How come we're changing this from a pointer?

Comment (vito) [reply atc/exec/try_step.go]: it's so the `ts.aborted = true` state persists through to the `Succeeded()` call

Comment (pivotal-jwinters) [reply atc/exec/try_step.go]: Why does wouldn't it get persisted in the pointer case?

Comment (vito) [reply atc/exec/try_step.go]: assuming you mean non-pointer case, `Run` is called with a copy of the struct as its receiver, so changes are only local to that function's copy

Comment (pivotal-jwinters) [reply atc/exec/try_step.go]: oh wow, I totally read this diff as going form ptr to value. haha don't mind me

Comment (josecv) [reply atc/exec/try_step.go]: I was under the impression this was necessary to mutate it later on, but if I'm wrong i can definitely fix it

Comment (pivotal-jwinters) [reply atc/exec/try_step.go]: no no its fine. I misread the diff and got confused for a second.

**Comment by cirocosta** [comment]
Thanks a lot, @josecv !

**Pull request merged**